ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/customercenter/InternalCustomerCenter.kt

```diff
@@ -0,0 +1,174 @@
+import com.revenuecat.purchases.paywalls.events.PaywallEvent
+import com.revenuecat.purchases.ui.revenuecatui.customercenter.data.CustomerCenterState
+import com.revenuecat.purchases.ui.revenuecatui.customercenter.data.CustomerCenterViewModel
+import com.revenuecat.purchases.ui.revenuecatui.customercenter.data.CustomerCenterViewModelFactory
+import com.revenuecat.purchases.ui.revenuecatui.customercenter.data.CustomerCenterViewModelImpl
+import com.revenuecat.purchases.ui.revenuecatui.data.PurchasesImpl
+import com.revenuecat.purchases.ui.revenuecatui.data.PurchasesType
+
+@Composable
+internal fun InternalCustomerCenter(
+    viewModel: CustomerCenterViewModel = getCustomerCenterViewModel(),
+) {
+    CustomerCenterScaffold {
+        when (val state = viewModel.state.collectAsState().value) {
+            is CustomerCenterState.Loading -> CustomerCenterLoading()
+            is CustomerCenterState.Error -> CustomerCenterError(state)
+            is CustomerCenterState.Success -> CustomerCenterLoaded(state)
+        }
+    }
+}
+
+@Composable
+private fun CustomerCenterScaffold(mainContent: @Composable () -> Unit) {
+    Column(
+        modifier = Modifier.fillMaxSize().padding(10.dp),
+        horizontalAlignment = Alignment.CenterHorizontally,
+        verticalArrangement = Arrangement.Center,
+    ) {
+        mainContent()
+    }
+}
+
+@Composable
+private fun CustomerCenterLoading() {
+    // CustomerCenter WIP: Add proper loading UI
+    Text("Loading...")
+}
+
+@Composable
+private fun CustomerCenterError(state: CustomerCenterState.Error) {
+    // CustomerCenter WIP: Add proper error UI
+    Text("Error: ${state.error}")
+}
+
+@Composable
+private fun CustomerCenterLoaded(state: CustomerCenterState.Success) {
+    // CustomerCenter WIP: Add proper success UI
+    Column {
+        Text("Customer Center config:", fontWeight = FontWeight.Bold, fontSize = 20.sp)
+        Text(state.customerCenterConfigDataString)
+    }
+}
+
+@Composable
+internal fun getCustomerCenterViewModel(
+    purchases: PurchasesType = PurchasesImpl(),
+    viewModel: CustomerCenterViewModel = viewModel<CustomerCenterViewModelImpl>(
+        factory = CustomerCenterViewModelFactory(purchases),
+    ),
+): CustomerCenterViewModel {
+    return viewModel
+}
+
+@Preview
+@Composable
+internal fun CustomerCenterLoadingPreview() {
+    CustomerCenterScaffold {
+        CustomerCenterLoading()
+    }
+}
+
+@Preview
+@Composable
+internal fun CustomerCenterErrorPreview() {
+    CustomerCenterScaffold {
+        CustomerCenterError(CustomerCenterState.Error(PurchasesError(PurchasesErrorCode.UnknownBackendError)))
+    }
+}
+
+@Preview
+@Composable
+internal fun CustomerCenterLoadedPreview() {
+    val viewModel = getCustomerCenterViewModel(previewPurchases)
+    InternalCustomerCenter(viewModel)
+}
+
+@OptIn(ExperimentalPreviewRevenueCatPurchasesAPI::class)
+private val previewPurchases = object : PurchasesType {
```

Comment:
I love that, thanks!

Reply:
One thing to note as I was thinking about this is that when we move forward, we might need to access the view model to perform operations such as performing an upgrade/downgrade, navigate to the subscription management page and so on. Once that happens we can either inject the view model in the corresponding composable, or pass the current one downwards through the composable tree (or create smaller view models in those). Probably the first option would be better, but in paywalls we are passing the main viewmodel downwards the composable tree right now. 

Nothing to do here right now, but I thought to mention it here, since, if we wanted to do the same thing we do for paywalls, we would need to pass a viewmodel around